packages/gestalt/src/MenuOption.js

```diff
@@ -0,0 +1,229 @@
+// @flow strict
+import React, { type Node } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import Badge from './Badge.js';
+import Box from './Box.js';
+import Flex from './Flex.js';
+import Link from './Link.js';
+import Text from './Text.js';
+import styles from './Touchable.css';
+import getRoundingClassName from './getRoundingClassName.js';
+import Icon from './Icon.js';
+import focusStyles from './Focus.css';
+import useFocusVisible from './useFocusVisible.js';
+
+type OptionObject = {|
+  label: string,
+  value: string,
+  subtext?: string,
+|};
+
+type OptionProps = {|
+  badgeText?: string,
+  children?: Node,
+  index: number,
+  option: OptionObject,
+  selected?: OptionObject | Array<OptionObject> | null,
+  handleSelect?: ({|
+    item: OptionObject,
+    event: SyntheticFocusEvent<HTMLInputElement>,
+  |}) => void,
+  hoveredItem: ?number,
+  isExternal?: boolean,
+  role?: 'option' | 'menuitem',
+  setHoveredItem: (number) => void,
+  setOptionRef: (?HTMLElement) => void,
+  shouldTruncate?: boolean,
+  textWeight?: 'bold' | 'normal',
```

Comment:
We want to avoid duplicating type definitions, so for these types that are passed directly to other components, we should import the type definition from that component (see the shared types between Box and Flex for examples, though the types can remain defined in Text rather than a separate types file)